.sandbox/provision/lifecycle-asg/common.sh

```diff
@@ -0,0 +1,141 @@
+#!/bin/bash
+
+: ${MAX_RETRIES:=30}
+: ${VOLUME_SIZE:=10}
+: ${DEVICE_NAME:="/dev/xvdf"}
+: ${SANDBOX_NAME_TAG:="SandboxManaged-SandboxName"}
+: ${SANDBOX_ID_TAG:="SandboxManaged-SandboxID"}
+: ${SANDBOX_ASG_TAG:="sandbox-asg"}
+
+function fatal() {
+    echo "$@" >&2
+    exit 1
+}
+
+function process_response() {
+    local response
+    if (( $# == 0 )) ; then
+        response="$(< /dev/stdin)"
+    else
+        response="$1"
+    fi
+
+    if [[ "$response" = "None" ]]; then
+        response=""
+    fi
+    echo "$response"
+}
+
+function get_volume_id() {
+    aws ec2 describe-volumes --filters Name=tag:"$SANDBOX_ID_TAG",Values="$SANDBOX_ID" --query 'Volumes[0].VolumeId' --output text | process_response
```

Comment:
While for `aws ... | something`, you will need to `set -o pipefail` at the beginning. Otherwise, failure of `aws ... ` part will be ignored, and still run `something`, unless this is what you want.